src/contracts/ParaSwapLiquiditySwapAdapterV2.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IPoolAddressesProvider} from '@aave/core-v3/contracts/interfaces/IPoolAddressesProvider.sol';
+import {DataTypes, ILendingPool} from 'aave-address-book/AaveV2.sol';
+import {IParaSwapAugustusRegistry} from './dependencies/paraswap/IParaSwapAugustusRegistry.sol';
+import {BaseParaSwapAdapter} from './base/BaseParaSwapAdapter.sol';
+import {ParaSwapLiquiditySwapAdapter} from './base/ParaSwapLiquiditySwapAdapter.sol';
+
+/**
+ * @title ParaSwapLiquiditySwapAdapterV2
+ * @notice ParaSwap Adapter to perform a swap of collateral from one asset to another.
+ * @dev It is specifically designed for Aave V2
+ * @author Aave Labs
+ **/
+contract ParaSwapLiquiditySwapAdapterV2 is ParaSwapLiquiditySwapAdapter {
+  /**
+   * @dev Constructor
+   * @param addressesProvider The address of the Aave PoolAddressesProvider contract
+   * @param pool The address of the Aave Pool contract
+   * @param augustusRegistry The address of the Paraswap AugustusRegistry contract
+   * @param owner The address of the owner
+   */
+  constructor(
+    IPoolAddressesProvider addressesProvider,
+    address pool,
+    IParaSwapAugustusRegistry augustusRegistry,
+    address owner
+  ) ParaSwapLiquiditySwapAdapter(addressesProvider, pool, augustusRegistry, owner) {
+    // Intentionally left blank
+  }
+
+  /// @inheritdoc BaseParaSwapAdapter
```

Comment:
shouldn't it be possible to abstact this into a AaveV2 and AaveV3 contract to just extend? seems a bit odd to implement this stuff for each adapter.